docs/how_to_publish_to_cmr_with_earthdata_varinfo.ipynb

```diff
@@ -0,0 +1,314 @@
+   "source": [
+    "### Get granule response and granule download URL with ```get_granules``` and ```get_granule_link``` for collection **M2I1NXASM**."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "granule_response = get_granules(concept_id='C1256535511-EEDTEST',\n",
+    "                                cmr_env=CMR_UAT,\n",
+    "                                auth_header=auth_header)\n",
+    "link = get_granule_link(granule_response)\n",
+    "link"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Then download the granule locally with `download_granule`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Saves granule to current directory or\n",
+    "# out_directory=/usr/given/path\n",
+    "download_granule(link, auth_header=auth_header)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Instantiate a ```VarInfoFromNetCDF4``` object for a local NetCDF-4 file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Instantiate a VarInfoFromNetCDF4 object for a local NetCDF-4 file.\n",
+    "var_info = VarInfoFromNetCDF4('MERRA2_400.inst1_2d_asm_Nx.20220130.nc4',\n",
+    "                              short_name='M2I1NXASM')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieve a dictionary of UMM-Var JSON records"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_dict = get_all_umm_var(var_info)\n",
+    "umm_var_dict"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish all the UMM-Var entries for M2I1NXASM"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "publish_all_umm_var('C1256535511-EEDTEST',\n",
+    "                    umm_var_dict,\n",
+    "                    auth_header=auth_header,\n",
+    "                    cmr_env=CMR_UAT)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish a single UMM-Var entry"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_entry = get_umm_var(var_info, var_info.get_variable('/TROPPV'))\n",
```

Comment:
Trying to summarise discussion with David here:

* This can be tricky if a user specifies the wrong variable path (e.g., a variable that doesn't exist), because that will mean the second argument is `None` and `get_umm_var` will fail.
* In practice people will most likely be iterating through the keys of variables in the `VarInfoFromNetCDF4` class, so shouldn't hit this.
* To maybe clear up confusion, though, the `var_info.get_variable('/TROPPV')` can be made it's own variable on a line above `umm_var_entry = get_umm_var(...)`, and there can be a bit of discussion about how the `VarInfoFromNetCDF4.get_variable` method can return `None` if a bad variable path is used.

Reply:
Thanks for clearing this up - I think the cell as it is now makes it really clear what might happen. Nice.